pkg/svcat/service-catalog/broker.go

```diff
@@ -120,11 +129,57 @@ func (sdk *SDK) Register(brokerName string, url string, opts *RegisterOptions) (
 		}
 
 	}
-	request := &v1beta1.ClusterServiceBroker{
+	if scopeOpts.Scope.Matches(ClusterScope) {
+		request := &v1beta1.ClusterServiceBroker{
+			ObjectMeta: v1.ObjectMeta{
+				Name: brokerName,
+			},
+			Spec: v1beta1.ClusterServiceBrokerSpec{
+				CommonServiceBrokerSpec: v1beta1.CommonServiceBrokerSpec{
+					CABundle:              caBytes,
+					InsecureSkipTLSVerify: opts.SkipTLS,
+					RelistBehavior:        opts.RelistBehavior,
+					RelistDuration:        opts.RelistDuration,
+					URL:                   url,
+					CatalogRestrictions: &v1beta1.CatalogRestrictions{
+						ServiceClass: opts.ClassRestrictions,
+						ServicePlan:  opts.PlanRestrictions,
+					},
+				},
+			},
+		}
+		if opts.BasicSecret != "" {
```

Comment:
Seems to me that all of the logic for populating the spec has been duplicated. I would prefer to see something like this

1. based on the scope, initialize a cluster or namespaced broker struct
2. populate the common spec in a single bit of code (in a function or inline)
3. based on the scope, persist the broker

Reply:
so i definitely agree that some dedupe can be done, but I don't think this much is possible due to how void interface pointers work in golang. I'll try to take a crack at it.